src/components/swaptab/index.js

```diff
@@ -252,23 +283,23 @@ class SwapTab extends React.Component {
               min={0.00000001}
               max={MAX}
               step={0.00000001}
-              error={error}
+              error={inputError}
               value={quoteAmount}
               onChange={e => this.updateBaseAmount(e)}
             />
             <DropDown
               defaultValue={quote}
               fields={currencies}
-              onChange={e => this.setState({ quote: e })}
+              onChange={e => this.checkValidPair(e, base)}
             />
           </View>
         </View>
         <View className={classes.next}>
           <Controls
             text={'Start swap'}
-            error={error}
+            error={error || inputError}
             onPress={error ? () => {} : () => this.shouldSubmit()}
-            errorText={'Invalid amount'}
+            errorText={inputError ? 'Invalid amount' : errorMessage}
```

Comment:
Do you think it is fine to have `Invalid amount` as default error? Would it be better to get rid of the `error` boolean and just have the string `errorText` that determines whether there is an error and sets the text of the button?

Reply:
@michael1011 My idea was that the input error would always override any other error.

Reply:
Yeah that sounds reasonable.